term/src/screen.rs

```diff
@@ -660,7 +660,10 @@ impl Screen {
         // changed by the scroll operation.  For normal newline at the bottom
         // of the screen based scrolling, the StableRowIndex does not change,
         // so we use the scroll region bounds to gate the invalidation.
-        if scroll_region.start != 0 || scroll_region.end as usize != self.physical_rows {
+        if scroll_region.start != 0
+            || scroll_region.end as usize != self.physical_rows
+            || !self.allow_scrollback
```

Comment:
I'm not sure about this change; it implies that the seqno's of the viewport are not being correctly invalidated by the main logic below for the common case. In particular, I'm concerned that if this is needed, it should probably be unconditional rather than scoped to the !allow_scrollback case.

The reason this is conditional is to avoid an additional O(screen-height) fixed overhead for every scroll operation.

Please translate your repro scenario from #6166 into a unit test in
https://github.com/wez/wezterm/blob/main/term/src/test/mod.rs that triggers the problem; I'd like to see what is happening to the seqno's of the lines being scrolled without attempting to fix it, then we can figure out what the right fix is.